src/components/Button.js

```diff
@@ -68,6 +69,9 @@ class Button extends React.Component {
     const mergedTheme = StyleUtils.mergeTheme(theme, primaryColor);
     const styles = this.styles(mergedTheme);
 
+    // We need to remove the styles prop from rest so we don't pass it to children.
+    delete rest.styles;
```

Comment:
Nice, I didn't think `delete` was fully supported in all browsers but it is.  👍 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete